### PR TITLE
[v3 2137] - Add more info to find out repair failures

### PR DESCRIPTION
**Pull request by ifraixedes**
What:  Add more information to the reported errors to find out how/where the repair process fails

Why: Because we are having troubles with repair and we want to find which is the part which causes the most of them

Please describe the tests: There is not change in the logic rather than, in majority, improving the error messages or wrapping errors, hence the existing tests are enough.
 
Please describe the performance impact: probably a bit, however we prefer to have more contextualized errors rather than having the minor improvement in performance.

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

**Comment by ifraixedes** [comment]
After the inspection of the code and with the minimal instructions that we have in the [ticket](https://storjlabs.atlassian.net/browse/V3-2137) that these changes should close, I don't see what else, except of abusing more of monkit tracing we can add, so please have a look and provide me feedback on what else should be added before this PR is moved out from draft.

File: pkg/storage/ec/client.go [hunk 1]
```diff
@@ -343,10 +343,15 @@ func (ec *ecClient) Get(ctx context.Context, limits []*pb.AddressedOrderLimit, p
 
 	rr, err = eestream.Decode(rrs, es, ec.memoryLimit, ec.forceErrorDetection)
 	if err != nil {
-		return nil, err
+		return nil, Error.Wrap(err)
+	}
+
+	ranger, err := eestream.Unpad(rr, int(paddedSize-size))
+	if err != nil {
```
Comment (wthorp) [inline pkg/storage/ec/client.go]: Nit: Error.Wrap returns nil if the input is nil.  I don't think you need this 'if' check.

File: satellite/metainfo/config.go [hunk 2]
```diff
@@ -57,6 +57,10 @@ func NewStore(logger *zap.Logger, dbURLString string) (db storage.KeyValueStore,
 		err = Error.New("unsupported db scheme: %s", driver)
 	}
 
+	if err != nil {
```
Comment (wthorp) [inline satellite/metainfo/config.go]: Again, this check is probably just extra code / against Go style guidelines.

File: storage/postgreskv/client.go [hunk 3]
```diff
@@ -89,12 +89,14 @@ func (client *Client) GetPath(ctx context.Context, bucket, key storage.Key) (_ s
 	row := client.pgConn.QueryRow(q, []byte(bucket), []byte(key))
 	var val []byte
 	err = row.Scan(&val)
-	if err == sql.ErrNoRows {
-		return nil, storage.ErrKeyNotFound.New(key.String())
-	}
 	if err != nil {
```
Comment (wthorp) [inline storage/postgreskv/client.go]: Nit: Again, this check is probably just extra code / against Go style guidelines.

Comment (wthorp) [inline storage/postgreskv/client.go]: Nit: Again, this check is probably just extra code / against Go style guidelines.

File: pkg/storage/ec/client.go (hunk 1 above)
Comment (ifraixedes) [reply pkg/storage/ec/client.go]: I do not like to abuse of it, but I think that's clear when nil is returned when no error than having to know that `Wrap` returns nil.

However I will change them

File: satellite/metainfo/config.go (hunk 2 above)
Comment (ifraixedes) [reply satellite/metainfo/config.go]: This one it's needed as the `log.Debug` should only happen if there is no error.

**Pull request merged**